Client/ChronoAdmin/client_admin.cpp

```diff
@@ -0,0 +1,267 @@
+/*BSD 2-Clause License
+
+Copyright (c) 2022, Scalable Computing Software Laboratory
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+   list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Created by Aparna on 01/12/2023
+*/
+
+#include <client.h>
+#include <common.h>
+#include <cassert>
+
+int main(int argc, char **argv) 
+{
+
+    std::string hostname;
+    std::string portnum;
+    std::string filename = "server_list";
+    ChronoLogRPCImplementation protocol;
+    std::string protocol_string; 
+    std::string server_uri;
+    std::vector<std::string> args;
+
+    if(argc != 7)
+    {
+       std::cout <<" ChronoAdmin usage : ./ChronoAdmin -protocol p -hostname h -port n"<<std::endl;
+       exit(-1);
+    }
+
+    for(int i=1;i<7;i++)
+    {
+	    std::string s(argv[i]);
+	    args.push_back(s);
+    }
+
+    bool end_program = false;
+
```

Comment:
Please, replace this  block with iteration loop over args[] array. The order of the command line arguments should not matter as long as key-value pair is present.  For example:  if( args[i] == "-p{rotocol}" and args[i+1] != end() and args[i+1]=="tcp" )  we have valid protocol and continue